testsuite/forge/src/backend/k8s/mod.rs

```diff
@@ -148,33 +156,79 @@ impl Factory for K8sFactory {
 
                 // We return early here if there are not enough PVs to claim.
                 check_persistent_volumes(
-                    kube_client,
+                    kube_client.clone(),
                     num_validators.get() + num_fullnodes,
                     existing_db_tag,
                 )
                 .await?;
             }
             // try installing testnet resources, but clean up if it fails
-            match install_testnet_resources(
-                self.kube_namespace.clone(),
-                num_validators.get(),
-                num_fullnodes,
-                format!("{}", init_version),
-                format!("{}", genesis_version),
-                genesis_modules_path,
-                self.use_port_forward,
-                self.enable_haproxy,
-                genesis_config_fn,
-                node_config_fn,
-            )
-            .await
-            {
-                Ok(res) => (Some(res.0), res.1, res.2),
-                Err(e) => {
-                    uninstall_testnet_resources(self.kube_namespace.clone()).await?;
-                    bail!(e);
-                },
-            }
+            let new_era = generate_new_era();
+            info!(
+                "Creating new era {} in namespace {}",
+                &new_era, &self.kube_namespace
+            );
+
+            let deploy_testnet_fut = Box::pin(async {
```

Comment:
fun tip: you can use futures::FutureExt here to replace the box pin with boxed()

```
Box::pin(async { ... })
```

Becomes 

```
async { ... }.boxed()
```